File: app/models/enums.py
"""Enum definitions matching DDL CHECK constraints."""

from enum import Enum


class DroneStatus(str, Enum):
    DISPONIVEL = "DISPONIVEL"
    EM_USO = "EM_USO"
    EM_MANUTENCAO = "EM_MANUTENCAO"
    BLOQUEADO = "BLOQUEADO"
    INATIVO = "INATIVO"


class BateriaStatus(str, Enum):
    DISPONIVEL = "DISPONIVEL"
    EM_USO = "EM_USO"
    CARREGANDO = "CARREGANDO"
    REPROVADA = "REPROVADA"
    DESCARTADA = "DESCARTADA"


class OrdemServicoStatus(str, Enum):
    RASCUNHO = "RASCUNHO"
    EM_ANALISE = "EM_ANALISE"
    APROVADA = "APROVADA"
    REJEITADA = "REJEITADA"
    CANCELADA = "CANCELADA"


class Prioridade(str, Enum):
    BAIXA = "BAIXA"
    MEDIA = "MEDIA"
    ALTA = "ALTA"
    CRITICA = "CRITICA"


class MissaoStatus(str, Enum):
    RASCUNHO = "RASCUNHO"
    PLANEJADA = "PLANEJADA"
    AGENDADA = "AGENDADA"
    EM_CHECKLIST = "EM_CHECKLIST"
    LIBERADA = "LIBERADA"
    EM_EXECUCAO = "EM_EXECUCAO"
    PAUSADA = "PAUSADA"
    CONCLUIDA = "CONCLUIDA"
    CANCELADA = "CANCELADA"
    ENCERRADA_TECNICAMENTE = "ENCERRADA_TECNICAMENTE"
    ENCERRADA_FINANCEIRAMENTE = "ENCERRADA_FINANCEIRAMENTE"


class ChecklistStatusGeral(str, Enum):
    PENDENTE = "PENDENTE"
    EM_PREENCHIMENTO = "EM_PREENCHIMENTO"
    CONCLUIDO = "CONCLUIDO"
    REPROVADO = "REPROVADO"
    APROVADO = "APROVADO"


class ItemChecklistStatus(str, Enum):
    PENDENTE = "PENDENTE"
    APROVADO = "APROVADO"
    REPROVADO = "REPROVADO"
    NAO_APLICAVEL = "NAO_APLICAVEL"


class Severidade(str, Enum):
    BAIXA = "BAIXA"
    MEDIA = "MEDIA"
    ALTA = "ALTA"
    CRITICA = "CRITICA"


class FinanceiroStatus(str, Enum):
    PENDENTE = "PENDENTE"
    EM_FATURAMENTO = "EM_FATURAMENTO"
    FATURADO = "FATURADO"
    RECEBIDO = "RECEBIDO"
    CANCELADO = "CANCELADO"


class DocumentoEntidade(str, Enum):
    DRONE = "DRONE"
    MANUTENCAO = "MANUTENCAO"
    USUARIO = "USUARIO"
    CLIENTE = "CLIENTE"
    PROPRIEDADE = "PROPRIEDADE"
    INSUMO = "INSUMO"
    MISSAO = "MISSAO"


class DocumentoStatus(str, Enum):
    ATIVO = "ATIVO"
    SUBSTITUIDO = "SUBSTITUIDO"
    VENCIDO = "VENCIDO"
    INATIVO = "INATIVO"

File: frontend/src/types/enums.ts
/** Enum string union types matching backend Python enums. */

export type DroneStatus =
  | 'DISPONIVEL'
  | 'EM_USO'
  | 'EM_MANUTENCAO'
  | 'BLOQUEADO'
  | 'INATIVO';

export type BateriaStatus =
  | 'DISPONIVEL'
  | 'EM_USO'
  | 'CARREGANDO'
  | 'REPROVADA'
  | 'DESCARTADA';

export type OrdemServicoStatus =
  | 'RASCUNHO'
  | 'EM_ANALISE'
  | 'APROVADA'
  | 'REJEITADA'
  | 'CANCELADA';

export type Prioridade =
  | 'BAIXA'
  | 'MEDIA'
  | 'ALTA'
  | 'CRITICA';

export type MissaoStatus =
  | 'RASCUNHO'
  | 'PLANEJADA'
  | 'AGENDADA'
  | 'EM_CHECKLIST'
  | 'LIBERADA'
  | 'EM_EXECUCAO'
  | 'PAUSADA'
  | 'CONCLUIDA'
  | 'CANCELADA'
  | 'ENCERRADA_TECNICAMENTE'
  | 'ENCERRADA_FINANCEIRAMENTE';

export type ChecklistStatusGeral =
  | 'PENDENTE'
  | 'EM_PREENCHIMENTO'
  | 'CONCLUIDO'
  | 'REPROVADO'
  | 'APROVADO';

export type ItemChecklistStatus =
  | 'PENDENTE'
  | 'APROVADO'
  | 'REPROVADO'
  | 'NAO_APLICAVEL';

export type Severidade =
  | 'BAIXA'
  | 'MEDIA'
  | 'ALTA'
  | 'CRITICA';

export type FinanceiroStatus =
  | 'PENDENTE'
  | 'EM_FATURAMENTO'
  | 'FATURADO'
  | 'RECEBIDO'
  | 'CANCELADO';

export type DocumentoEntidade =
  | 'DRONE'
  | 'MANUTENCAO'
  | 'USUARIO'
  | 'CLIENTE'
  | 'PROPRIEDADE'
  | 'INSUMO'
  | 'MISSAO';

export type DocumentoStatus =
  | 'ATIVO'
  | 'SUBSTITUIDO'
  | 'VENCIDO'
  | 'INATIVO';
